core/src/graph/graph.rs

```diff
@@ -569,15 +609,75 @@ impl Graph {
 	/// Determine if page is full
 	///  aggressive:false -> use a simple heuristic based on the number of connections
 	///  aggressive:true  -> do actual compression to determine resulting actual page size
-	pub fn is_page_full(&self, page: &GraphPage, aggressive: bool) -> bool {
+	pub fn try_add_connection_to_page(
```

Comment:
nit: Not a blocker but I think we should pass in the encryption key instead of fetching it here again. One way to facilitate tests for this might be to create a second method just for tests `#[cfg(test)]` (without encryption key) that gets the encryption key from `user_key_manager`

Reply:
Done.